jni/src/faiss_wrapper.cpp

```diff
@@ -624,6 +697,57 @@ jbyteArray knn_jni::faiss_wrapper::TrainIndex(knn_jni::JNIUtilInterface * jniUti
     return ret;
 }
 
+jbyteArray knn_jni::faiss_wrapper::TrainBinaryIndex(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jobject parametersJ,
```

Comment:
This also need to be refactored to reduce code duplication.
We have GH issue for query part. https://github.com/opensearch-project/k-NN/issues/1773

Reply:
Will have another PR to the refactor.